lib/api/1.1/northbound/nodes.js

```diff
@@ -569,9 +577,12 @@ function nodesRouterFactory (
     router.delete('/nodes/:identifier/workflows/active', rest(function (req) {
         return waterline.nodes.needByIdentifier(req.params.identifier)
         .then(function (node) {
-            return taskGraphProtocol.cancelTaskGraph({ target: node.id });
+            return workflowApiService.findActiveGraphForTarget(node.id);
+        })
+        .then(function(graph) {
+            return workflowApiService.cancelTaskGraph(graph.instanceId);
```

Comment:
Good question, I hadn't noticed this edge case. In the case that the graph finishes after the `findActiveGraphForTarget` call but before the `cancelTaskGraph` call, the user will get back a 204 No Content (just tested this out). Just brainstorming, perhaps we want to tack on another step that returns the instanceId no matter what on success, or do something else?